### PR TITLE
Validation filter result

### PR DESCRIPTION
**Pull request by woksin**
Based on PR #366, pull that one in first.

Start on simplifying the filters. This is an ongoing process. More PRs will follow.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1140654762299587/1170011480522006)

### CODE REVIEW

File: Source/EventHorizon/Producer/Filter/PublicFilterDefinition.cs [hunk 1]
```diff
@@ -27,5 +27,8 @@ public PublicFilterDefinition(StreamId sourceStream, StreamId targetStream)
 
         /// <inheritdoc/>
         public StreamId TargetStream { get; }
+
+        /// <inheritdoc/>
+        public bool IsPersistable => false;
```
Comment (einari) [inline Source/EventHorizon/Producer/Filter/PublicFilterDefinition.cs:32]: You could have an interface `ICanBePersisted` instead - marker interface. Use this for checking if it should be persisted (`if (filterDefinition is ICanBePersisted) {}`)

**Comment by einari** [comment]
suggestion: Use MongoDB Class Maps to represent the persistence: http://mongodb.github.io/mongo-csharp-driver/2.0/reference/bson/mapping/

File: Source/Events.Processing/Filters/FilterRegistry.cs [hunk 2]
```diff
@@ -24,85 +20,57 @@ public class FilterRegistry : IFilterRegistry
...
+            if (!validationResult.Succeeded)
             {
-                _logger.Error(ex, $"No persisted filter repository for filter definition of type {typeof(TDefinition).FullName}");
+                var message = $"Filter '{filter.Definition.TargetStream}' failed validation: {validationResult.FailureReason}";
+                _logger.Debug(message);
+                throw new CouldNotRegisterFilter(filter.Definition, message);
             }
 
-            if (repository != null)
+            if (filter.Definition.IsPersistable)
```
Comment (einari) [inline Source/Events.Processing/Filters/FilterRegistry.cs:62]: Use the `ICanBePersisted` interface to check for if it should be persisted

**Comment by einari** [comment]
Suggest also looking into the Concept serializer: https://github.com/dolittle-extensions/ReadModels.MongoDB/blob/master/Source/ConceptSerializationProvider.cs

File: Source/EventHorizon/Producer/Filter/PublicFilterDefinition.cs (hunk 1 above)
Comment (einari) [reply Source/EventHorizon/Producer/Filter/PublicFilterDefinition.cs:32]: Or...  `IPersistedFilterDefinition`

**Comment by woksin** [comment]
Closing this. This is coming as a part of another PR.

**Pull request closed without merging**